chalk/shapes/arc.py

```diff
@@ -0,0 +1,162 @@
+        return p2.length
+
+    @property
+    def rot(self) -> float:
+        t2 = tx.remove_translation(self.t)
+        return tx.apply_p2_affine(t2, unit_x).angle
+
+    @property
+    def center(self) -> P2:
+        "Real center"
+        return tx.apply_p2_affine(self.t, P2(0, 0))
+
+    def apply_transform(self, t: tx.Affine) -> ArcSegment:  # type: ignore
+        return ArcSegment(self.angle, self.dangle, t * self.t)
+
+    def get_trace(self, t: tx.Affine = Ident) -> Trace:
+        "Trace is done as simple arc and transformed"
+        angle0_deg = self.angle
+        angle1_deg = self.angle + self.dangle
+
+        def f(p: P2, v: V2) -> List[float]:
+            ray = Ray(p, v)
+            return sorted(
+                [
+                    d / v.length
+                    for d in ray_circle_intersection(ray, 1)
+                    if is_in_mod_360(
+                        ((d * v) + P2.polar(self.angle, 1)).angle,
+                        min(angle0_deg, angle1_deg),
+                        max(angle0_deg, angle1_deg),
+                    )
+                ]
+            )
+
+        return Trace(f).apply_transform(self.t)
+
+    def get_envelope(self, t: tx.Affine = Ident) -> Envelope:
+        "Trace is done as simple arc and transformed"
+        angle0_deg = self.angle
+        angle1_deg = self.angle + self.dangle
+
+        v1 = V2.polar(angle0_deg, 1)
+        v2 = V2.polar(angle1_deg, 1)
+
+        def wrapped(d: V2) -> float:
+            is_circle = abs(angle0_deg - angle1_deg) >= 360
+            if is_circle or is_in_mod_360(
+                d.angle,
+                min(angle0_deg, angle1_deg),
+                max(angle0_deg, angle1_deg),
+            ):
+                # Case 1: P2 at arc
+                return 1 / d.length
+            else:
+                # Case 2: P2 outside of arc
+                x: float = max(d.dot(v1), d.dot(v2))
+                return x
+
+        return Envelope(wrapped).apply_transform(self.t)
+
+    @staticmethod
+    def arc_between(p: P2, q: P2, height: float) -> SegmentLike:
+
+        h = abs(height)
+        if h < 1e-3:
+            return Segment(p, q)
+        d = (q - p).length
+        # Determine the arc's angle θ and its radius r
+        θ = math.acos((d**2 - 4.0 * h**2) / (d**2 + 4.0 * h**2))
+        r = d / (2 * math.sin(θ))
+
+        if height > 0:
+            # bend left
+            φ = +math.pi / 2
+            dy = r - h
+            flip = 1
+        else:
+            # bend right
+            φ = -math.pi / 2
+            dy = h - r
+            flip = -1
+
+        diff = q - p
+        ret = (
+            ArcSegment(flip * -from_radians(θ), flip * 2 * from_radians(θ))
+            .scale(r)
+            .rotate_rad(φ)
+            .translate(d / 2, dy)
+        )
+
+        ret = ret.rotate(-diff.angle).translate_by(p)
+        return ret
+
+    def reflect_y(self) -> ArcSegment:
+        assert False
```

Comment:
I've noticed that the `reflect` transformations are not defined. Is there a particular reason for this? Isn't the default behavior (specified by `Transformable`) correct?

Reply:
I got a bit stuck here. The problem is that reflection changes that sign of `dangle` I wasn't sure how to preserve this information in my parameterization. Given that reflection of a segment within a path (as opposed to the full path) seemed rare, I decided not to handle it.

Reply:
Okay, I think I see what's the problem. Maybe it can be fixed if we take into account the sign of the determinant of the transformation matrix (a negative determinant means that the direction of the arc changes); something along the following lines:
```diff
diff --git a/chalk/backend/svg.py b/chalk/backend/svg.py
index 72f382e..f02edbf 100644
--- a/chalk/backend/svg.py
+++ b/chalk/backend/svg.py
@@ -110,7 +110,7 @@ class ToSVGShape(ShapeVisitor[BaseElement]):
         elif isinstance(seg, ArcSegment):
             "https://www.w3.org/TR/SVG/implnote.html#ArcConversionCenterToEndpoint"
             f_A = 1 if abs(seg.dangle) > 180 else 0
-            f_S = 1 if seg.dangle > 0 else 0
+            f_S = 1 if seg.t.determinant * seg.dangle > 0 else 0
             return f"A {seg.r_x} {seg.r_y} {seg.rot} {f_A} {f_S} {seg.q.x} {seg.q.y}"
         else:
             assert False, "not a known segment"
diff --git a/chalk/backend/tikz.py b/chalk/backend/tikz.py
index 725c5c1..8a8ae8a 100644
--- a/chalk/backend/tikz.py
+++ b/chalk/backend/tikz.py
@@ -110,9 +110,9 @@ class ToTikZShape(ShapeVisitor[PyLatexElement]):
             start = (seg.p - seg.center).angle
             end = (seg.q - seg.center).angle
-            if seg.dangle < 0 and end > start:
+            if seg.t.determinant * seg.dangle < 0 and end > start:
                 end = end - 360
-            if seg.dangle > 0 and end < start:
+            if seg.t.determinant * seg.dangle > 0 and end < start:
                 end = end + 360
             end_ang = end - seg.rot
             pts._arg_list.append(
```
This change seem to produce the desired results on a simple example (I have defined the `stroke` method on `ArcSegment`):
```python
from chalk import *
from chalk.shapes import ArcSegment

dia = cat(
    [
        ArcSegment(180, 135).stroke().show_origin(),
        ArcSegment(180, 135).scale_x(-1).stroke().show_origin(),
        ArcSegment(180, 135).stroke().scale_x(-1).show_origin(),
    ],
    unit_x,
    sep=0.5
)

dia.render("examples/output/test_arc.png")
dia.render_svg("examples/output/test_arc.svg")
dia.render_pdf("examples/output/test_arc.pdf")
```
![test_arc](https://user-images.githubusercontent.com/819256/183860598-ad1953ee-9030-4b07-8e41-bb5c107ab02f.svg)

Reply:
What a nice solution! Added.